### ISSUE TITLE
KITTI Visualization Results

### ISSUE THREAD

**Issue by 757787182** [Sungmin-Woo/ProDepth#9]
Nice Work！！！I am using the KITTI dataset provided by you for visualization and comparative experiments, and I have found that the provided model does not have the ability to handle dynamic objects. Is there a problem with this? Can you help me resolve this issue?
![image](https://github.com/user-attachments/assets/b64e9b8a-c98b-468d-af19-d92a5b068bb4)
this is mono depth error on AbsRel
![image](https://github.com/user-attachments/assets/22ed04bc-38fc-49e0-a315-96e413992b4b)
this is mvs

**Comment by Sungmin-Woo** [comment]
The original code contained some errors, and the updated code with necessary modifications has now been re-uploaded. Please use the updated code and let me know if the issue persists :)

**Comment by 757787182** [comment]
> The original code contained some errors, and the updated code with necessary modifications has now been re-uploaded. Please use the updated code and let me know if the issue persists :)

Could you please provide the model weights of the KITTI dataset again?
![image](https://github.com/user-attachments/assets/a65af3aa-6632-49a9-8cff-3cca3b759efe)

**Comment by 757787182** [comment]
> The original code contained some errors, and the updated code with necessary modifications has now been re-uploaded. Please use the updated code and let me know if the issue persists :)

Could you please provide the model weights of the KITTI dataset again?these is the KITTI benchmark scores?
![image](https://github.com/user-attachments/assets/c71417f4-272a-4619-a604-c0db28ad726e)

**Comment by 757787182** [comment]
> The original code contained some errors, and the updated code with necessary modifications has now been re-uploaded. Please use the updated code and let me know if the issue persists :)
Could you please provide the model weights of the KITTI dataset again?these  is the KITTI benchmark scores?

![image](https://github.com/user-attachments/assets/fa097038-ee20-4dd4-80e9-724e4b1e07f3)

**Comment by Sungmin-Woo** [comment]
Since the training code, not the inference code, has been modified, the provided weights can still be used as they are. On the KITTI benchmark, ProDepth achieves an AbsRel of approximately 0.059 when using the improved ground truth depth map, so it seems likely that you conducted the evaluation with the improved GT. Please refer to Table 9 in the arXiv paper for further details.

**Comment by 757787182** [comment]
> Since the training code, not the inference code, has been modified, the provided weights can still be used as they are. On the KITTI benchmark, ProDepth achieves an AbsRel of approximately 0.059 when using the improved ground truth depth map, so it seems likely that you conducted the evaluation with the improved GT. Please refer to Table 9 in the arXiv paper for further details.

Thank you very much for your explanation. I hope to obtain the KITTI model in Table 1 (I couldn't find the download location) for the qualitative analysis of my paper. Could you please provide it? thank you.
![image](https://github.com/user-attachments/assets/7607a5c0-1f4a-4849-bf18-67201306e39c)

**Comment by 757787182** [comment]
> Since the training code, not the inference code, has been modified, the provided weights can still be used as they are. On the KITTI benchmark, ProDepth achieves an AbsRel of approximately 0.059 when using the improved ground truth depth map, so it seems likely that you conducted the evaluation with the improved GT. Please refer to Table 9 in the arXiv paper for further details.

Hello, I have only recently started to get involved in this field. I'm not quite clear about what you mean by "KITTI evaluation on improved ground truth". Do you mean that the model in Table 1 is used, but with a denser ground truth depth? I have always been using the standard KITTI ground truth data, and there is a certain deviation between the results I have tested and those in your Table 9. I can be certain that the model weights you have improved are not the KITTI model weights in Table 1.

**Comment by Sungmin-Woo** [comment]
The newly updated code only involves changes in the training strategy, while the model structure remains the same. Therefore, using the KITTI model weights provided in the repository (the model you pointed out) for performance evaluation should yield the correct results. I just verified this by cloning the repository and running it from scratch, and I was able to confirm that it achieves 0.086, as expected:

![20250106_173141](https://github.com/user-attachments/assets/b5f4055e-546f-4cec-bc84-a00a9cf79d03)


As you mentioned, Table 1 presents the evaluation results using the standard KITTI GT data, whereas Table 9 shows the results using the denser GT data. If you used the standard GT values, the AbsRel should indeed be 0.086.

What seems odd is that the performance you shared earlier is 0.057, which is significantly better than the AbsRel 0.086 achieved by the provided weights. This leads me to suspect that there might be an issue with how the evaluation metrics are being calculated. Alternatively, since the metric calculation code in the repository is fine, if you haven't changed the code, the issue might be with the data.

I hope you can double-check everything from the beginning to make sure there are no mistakes :)

**Comment by 757787182** [comment]
> The newly updated code only involves changes in the training strategy, while the model structure remains the same. Therefore, using the KITTI model weights provided in the repository (the model you pointed out) for performance evaluation should yield the correct results. I just verified this by cloning the repository and running it from scratch, and I was able to confirm that it achieves 0.086, as expected:
> 
> ![20250106_173141](https://private-user-images.githubusercontent.com/66046365/400356637-b5f4055e-546f-4cec-bc84-a00a9cf79d03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxNTQzNjAsIm5iZiI6MTczNjE1NDA2MCwicGF0aCI6Ii82NjA0NjM2NS80MDAzNTY2MzctYjVmNDA1NWUtNTQ2Zi00Y2VjLWJjODQtYTAwYTljZjc5ZDAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA2VDA5MDEwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1NDU5MzNhMzA5YmVmMjBiZThmOTIyZDg1Nzc2OGZiNzYzNTdjNWY3ZWU0MTI4NWNmNjUzYWU1YjY2MDk2Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.70R3Kjojyhnup5CBokKPhMPPaacPeMCNYBieDeeZx6k)
> 
> As you mentioned, Table 1 presents the evaluation results using the standard KITTI GT data, whereas Table 9 shows the results using the denser GT data. If you used the standard GT values, the AbsRel should indeed be 0.086.
> 
> What seems odd is that the performance you shared earlier is 0.057, which is significantly better than the AbsRel 0.086 achieved by the provided weights. This leads me to suspect that there might be an issue with how the evaluation metrics are being calculated. Alternatively, since the metric calculation code in the repository is fine, if you haven't changed the code, the issue might be with the data.
> 
> I hope you can double-check everything from the beginning to make sure there are no mistakes :)

I re-downloaded your model because your model test requires lite-mono-8m-pretrain.pth. I deleted the code for loading the pre-trained model because the model training doesn't need the support of pre-training. I didn't replace any of your evaluation metrics. Could you please send the model you used for testing to 757787182@qq.com? thank you
![image](https://github.com/user-attachments/assets/799ea222-c9ba-4835-a77e-b03c803679ae)

**Comment by 757787182** [comment]
> The newly updated code only involves changes in the training strategy, while the model structure remains the same. Therefore, using the KITTI model weights provided in the repository (the model you pointed out) for performance evaluation should yield the correct results. I just verified this by cloning the repository and running it from scratch, and I was able to confirm that it achieves 0.086, as expected:
> 
> ![20250106_173141](https://private-user-images.githubusercontent.com/66046365/400356637-b5f4055e-546f-4cec-bc84-a00a9cf79d03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxNTQzNjAsIm5iZiI6MTczNjE1NDA2MCwicGF0aCI6Ii82NjA0NjM2NS80MDAzNTY2MzctYjVmNDA1NWUtNTQ2Zi00Y2VjLWJjODQtYTAwYTljZjc5ZDAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA2VDA5MDEwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1NDU5MzNhMzA5YmVmMjBiZThmOTIyZDg1Nzc2OGZiNzYzNTdjNWY3ZWU0MTI4NWNmNjUzYWU1YjY2MDk2Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.70R3Kjojyhnup5CBokKPhMPPaacPeMCNYBieDeeZx6k)
> 
> As you mentioned, Table 1 presents the evaluation results using the standard KITTI GT data, whereas Table 9 shows the results using the denser GT data. If you used the standard GT values, the AbsRel should indeed be 0.086.
> 
> What seems odd is that the performance you shared earlier is 0.057, which is significantly better than the AbsRel 0.086 achieved by the provided weights. This leads me to suspect that there might be an issue with how the evaluation metrics are being calculated. Alternatively, since the metric calculation code in the repository is fine, if you haven't changed the code, the issue might be with the data.
> 
> I hope you can double-check everything from the beginning to make sure there are no mistakes :)

Thank you for your patient reply. I have found the problem. I compared the predicted depth saved by my model with the depth predicted by your model. I'm very sorry for the trouble I caused you.

![image](https://github.com/user-attachments/assets/196e94f1-a4c7-4404-9341-8c7054f599e9)

**Comment by Sungmin-Woo** [comment]
No problem, I'm just glad to hear it's been resolved.
Good luck with your research!
